### ISSUE TITLE
Merge PR script from external contributor with code signing

### ISSUE THREAD

**Issue by oskarth** [status-im/status-react#2636]
### User Story

As a developer, I want all commits that make it into develop to be code signed by a core contributor's PGP key so that I know the certified source of changes introduced in our code base.

### Description

*Type*: Feature/Tooling

*Summary*: Write a script that automate most parts of merging a PR from an external contributor while still signing the commit from the command line.

For context: https://github.com/status-im/status-react/pull/2611 for merge script for core contributors, and https://wiki.status.im/Code_signing_commits_into_develop.

#### Expected behavior
Something like: `./merge-external-pr.sh REMOTE BRANCH` (some manual action) and then commit shows up in `develop` branch, GPG signed by person running script and authored by person submitting initial commit.

While it'd be nice if PR was created automatically it isn't necessary. For example script can prompt you to do action in UI and then continue once you confirmed X is done. Perhaps it ends up being two scripts, as PR needs to be reviewed with our current Github trust model.

Preferably written in a similar style as `merge-sh` (short, self-contained, documented). Optional dependency: something that simplifies pull request dealings (such as hub by Github).

Nice to have but very optional: flow that makes external PRs appear as merged as opposed to closed. No idea how this is possibly though without introducing other branches.

#### Actual behavior
Either no code signing (or Github only) or a quite cumbersome manual process as outlined in Wiki.

**Comment by status-open-bounty** [comment]
Balance: 0.000000 ETH
Tokens: SNT: 4000.00
Contract address: [0xbe4fd5b148bd1e38cc0d65d30db2237d597e5914](https://etherscan.io/address/0xbe4fd5b148bd1e38cc0d65d30db2237d597e5914)
Network: Mainnet
Paid to: dottedmag
Visit [https://openbounty.status.im](https://openbounty.status.im) to learn more.

**Comment by dottedmag** [comment]
I had a look at this issue, made a crude version of a script, and now would like to understand the typical workflow to make it more usable.

Do maintainers add a remote, fetch it and review locally, and then merge, or do they review code on GitHub and want the script only for a merge stop?

**Comment by oskarth** [comment]
@dottedmag Thanks! I think it's more common for core contributors to review code in Github and then only run merge script from CLI, but some might prefer to fetch it and review locally (@rasom?). Whatever is easiest would be great to get started :)

**Comment by dottedmag** [comment]
A case with a local branch for is easier — while information for looking up PR details is one `curl` away, Bash does not have JSON parser, `grep`-ing result is brittle, and `hub(1)` is not especially helpful either.

**Comment by oskarth** [comment]
In that case, go for it! Anything than what we currently have is an improvement :) Would you mind including instructions for how a normal core contributor should deal with an external PR in your upcoming PR? adding remotes etc.

**Comment by dottedmag** [comment]
No problem, where should these instructions go?

**Comment by oskarth** [comment]
In the PR and as comments in the script would be great!

**Comment by dottedmag** [comment]
The current WiP is in the PR — is this a right direction to go?
